src/lib/common/sol-bus.c

```diff
@@ -383,9 +383,8 @@ sol_bus_client_free(struct sol_bus_client *client)
     if (!client)
         return;
 
-    destroy_client(client);
-
     sol_ptr_vector_remove(&_ctx.clients, client);
+    destroy_client(client);
```

Comment:
On Fri, 06 May 2016 10:56:39 -0700, Gustavo Sverzut Barbieri wrote:
> >      sol_ptr_vector_remove(&_ctx.clients, client);
> > +    destroy_client(client);
> 
> makes no sense to me since nobody changes the numeric value, it's basically an "unsigned long int".

Yes, in practice, but there are all sort of things in the standard that
justify saying that after free(), the value of the pointer is invalid.
It's the same reason why arithmetic of pointers that don't all point to
the same object is invalid.